File: TestProject1/Class/UnitTest_ClsCmdExec.cs
using CmnClsLib.Module;
using CmnClsLib.Interface;
using CmnClsLib.Class;
using Xunit;
using Assert = Xunit.Assert;

namespace TestProject1.Class
{
    public class UnitTest_ClsCmdExec
    {
        private static ClsLogger _logger = new();
        private ClsCmdExec _cmdExec = new(_logger);

        // --------------------------------------------------------------------
        // ExecuteThread()
        // --------------------------------------------------------------------
        [Fact]
        public void Execute_ipconfigが正常終了すること()
        {
            _cmdExec.CmdPath = "" + System.Environment.GetEnvironmentVariable("ComSpec");
            _cmdExec.CmdArgs = "/c ipconfig /all";
            Assert.Equal(0, _cmdExec.ExecuteThread(3));
        }
        [Fact]
        public void Execute_存在しないコマンドを実行した場合異常終了すること()
        {
            _cmdExec.CmdPath = "" + System.Environment.GetEnvironmentVariable("ComSpec");
            _cmdExec.CmdArgs = "/c ifconfig /all";
            Assert.Equal(1, _cmdExec.ExecuteThread(3));
        }
    }
}

File: src/test/java/tool/cmnclslib/cls/UnitTest_ClsCmdExec.java
package tool.cmnclslib.cls;

import org.junit.Before;
import org.junit.Test;
import tool.cmnclslib.mdl.MdlApp;
import static org.junit.Assert.*;

/**
 * ClsCmdExec の単体テストクラスです。
 */
public class UnitTest_ClsCmdExec {

    private ClsLogger logger = new ClsLogger();
    private ClsCmdExec cmdExec;

    @Before
    public void setUp() {
        cmdExec = new ClsCmdExec(logger);
    }

    @Test
    public void execute_標準コマンドが正常終了すること() {
        if (MdlApp.isWindows()) {
            String comSpec = System.getenv("ComSpec");
            cmdExec.setCmdPath(comSpec != null && !comSpec.isEmpty() ? comSpec : "cmd.exe");
            cmdExec.setCmdArgs("/c echo test");
        } else {
            cmdExec.setCmdPath("sh");
            cmdExec.setCmdArgs("-c echo test");
        }
        assertEquals(0, cmdExec.executeThread(3));
    }

    @Test
    public void execute_存在しないコマンドを実行した場合異常終了すること() {
        if (MdlApp.isWindows()) {
            String comSpec = System.getenv("ComSpec");
            cmdExec.setCmdPath(comSpec != null && !comSpec.isEmpty() ? comSpec : "cmd.exe");
            cmdExec.setCmdArgs("/c non_existent_command_xyz_12345");
            assertNotEquals(0, cmdExec.executeThread(3));
        } else {
            cmdExec.setCmdPath("sh");
            cmdExec.setCmdArgs("-c non_existent_command_xyz_12345");
            assertNotEquals(0, cmdExec.executeThread(3));
        }
    }
}
